mesh/mesh.go

```diff
@@ -249,21 +252,58 @@ func (msh *Mesh) getProcessedLayer() ProcessedLayer {
 	return msh.processedLayer
 }
 
-func (msh *Mesh) setProcessedLayer(layerID types.LayerID, hash types.Hash32) {
+func (msh *Mesh) setProcessedLayerFromRecoveredData(pLayer *ProcessedLayer) {
 	msh.mutex.Lock()
 	defer msh.mutex.Unlock()
-	if msh.processedLayer.ID.After(types.NewLayerID(0)) && msh.processedLayer.ID.Add(1) != layerID {
-		msh.With().Error("could not set processed layer",
-			log.FieldNamed("old_processed_layer", msh.processedLayer.ID),
-			log.FieldNamed("new_processed_layer", layerID))
+	msh.processedLayer = *pLayer
+	msh.Event().Info("processed layer set from recovered data", pLayer.ID, log.String("layer_hash", pLayer.Hash.Hex()))
+}
+
+func (msh *Mesh) setProcessedLayer(layer *types.Layer) {
+	msh.mutex.Lock()
+	defer msh.mutex.Unlock()
+	if !layer.Index().After(msh.processedLayer.ID) {
+		msh.With().Warning("attempt to set processed layer to an older layer",
```

Comment:
if we try to set a processed layer with new blocks than perhaps we should allow this and also calculate aggregated hashes for all subsequent layers as well.

Reply:
done. also added test